tests/conftest.py

```diff
@@ -202,11 +199,21 @@ def applySignature(self, signatureName, address):
         return contract
 
     def resetSnapshot(self):
-        self.chain.block = self.originalBlock
-        self.chain.revert(self.snapshot)
-        self.chain.head_state = self.originalHead
-        self.contracts = deepcopy(self.originalContracts)
+        self.resetToSnapshot(self.captured)
 
+    def createSnapshot(self):
+        self.chain.mine(1)
+        self.captured = {'block': self.chain.block, 'head_state': self.chain.head_state, 'contracts' : deepcopy(self.contracts), 'snapshot': self.chain.snapshot()}
+        return self.captured
+
+    def resetToSnapshot(self, captured):
+        if len(captured) > 0:
+            self.chain.block = captured['block']
+            self.chain.revert(captured['snapshot'])
+            self.chain.head_state = captured['head_state']
+            self.contracts = deepcopy(captured['contracts'])
+        else:
+            self.createSnapshot()
```

Comment:
Recommend removing this `if/else` block as creating a snapshot if someone tries to reset is the wrong behavior.  It may be good to do
`if len(captured) <= 4: raise Exception("invalid snapshot")`
(or however you throw in Python)